src/AlgebraicGeometry/Schemes/ProjectiveVariety/Objects/Types.jl

```diff
@@ -0,0 +1,16 @@
+@attributes mutable struct ProjectiveVariety{BaseRing<:Field, GradedRingType} <: AbsProjectiveVariety{BaseRing, GradedRingType}
+  X::AbsProjectiveScheme
+
+  function ProjectiveVariety(X::AbsProjectiveScheme{S, T}; check::Bool) where {S, T}
+    if check
+      S <:QQField || error("varieties must be geometrically integral, but we test this only over QQ at the moment, disable this check if you know the variety is geometrically integral or proceed at your own risk")
+      is_geometrically_integral(X) || error("varieties must be geometrically integral")
+    else
+      set_attribute!(X, :is_geometrically_integral => true)
+      set_attribute!(X, :is_integral => true)
+      set_attribute!(X, :is_geometrically_reduced => true)
+      set_attribute!(X, :is_reduced => true)
```

Comment:
Alternatively once can override the getters for the corresponding abstract type. That saves the time for the dictionary lookup.

Reply:
It has the drawback that we do not have
`has_attribute(:is_reduced)` to check if the scheme knows that it is reduced. 
We could introduce an interface for this of course e.g. 
`has_is_reduced` .